pkg/blueprint/disk_customizations.go

```diff
@@ -687,3 +687,30 @@ func CheckDiskMountpointsPolicy(partitioning *DiskCustomization, mountpointAllow
 
 	return nil
 }
+
+// GetDiskMinSize calculates the size required to fit all the disk
+// customization options. It returns the Disk MinSize or the sum of all the
+// Partitions, whichever is larger.
+func (c *Customizations) GetDiskMinSize() uint64 {
+	if c == nil || c.Disk == nil {
+		return 0
+	}
+
+	minDiskSize := c.Disk.MinSize
+
+	var sum uint64
+	for _, part := range c.Disk.Partitions {
+		switch part.Type {
+		case "plain", "btrfs", "":
```

Comment:
It would be nice to be confident that all that all disk types are covered (now and in the future when we potentially add one)? Might be that we just  using `default:` here and hope for the best? Or panic for unknown ones (well, error would be better)